controllers/auth.js

```diff
@@ -51,14 +51,44 @@ module.exports = {
     };
   },
   register: async (req, res) => { 
-    try { 
-      // TODO: User registration
 
-    } catch (error) { 
-      return res.status(500).json({ 
-        message: error,
-      });
-    };
+    // creates uuid to use as user's ref number with the help of uuid npm
+    const uuid = uuidv4();
+    const { first_name, last_name, email, password } = req.body
+
+    // saltRound for bcrypt
+    const saltRounds = 10;
+
+    //checks if user already exists
+    connection.query('select * from users where email = ?', [email], (err, results) => {
+      // if no user exists create hashed password and add to user table
+      if (results[0] === undefined) {        
+        // creates hashed password with the help of bcrypt npm
+        bcrypt.hash(password, saltRounds, function (err, hash) {
+          if (err) {
+            return res.status(500).send(err)
+          }
+          else {
+            let user = [uuid, email, hash, first_name, last_name]
+            // adds user to the user table
+            connection.query('insert into users (uuid, email, password, first_name, last_name) values (?)', [user], async (error, results) => {
+              if (err) {
+                return res.status(500).send(err)
+              }
+              return res.status(200).json({
+                saved: true,
+                uuid: uuid,
+              })
+            });
+          }
+        });
+      // if user does exist send back message
+      } else {
+        return res.status(200).json({
+          saved: false,
```

Comment:
I think we should add a `message` key here to send back a reason that can be displayed to the user--it would be easier to access the reason in the response and update this on the page rather than keep a list of possible return reasons client side. Thoughts?

Reply:
Originally I did have a message coming back but got rid of it because I didn't see the point. I like your idea that we would automatically display that message to the user.  I've gone ahead and add it to the json and put the message in the alert. I suppose right now there is only one reason they can't register which if the email is duplicated so we don't necessarily need it until we have alternate reasons they can't register.  But I think it is better to have from the beginning. I've gone ahead and add it to the json and put the message in the alert.